### ISSUE TITLE
character limit

### ISSUE THREAD

**Issue by zxz41** [Metastruct/garrysmod-chatsounds#42]
What do you do if a sound exceeds the character limit (255)?

The qoute is:
"no no im still not feeling it i want this achievement to have meant something it has to be a true reward for valiant effort i want to see some hustle stanley i want to see commitment a willingness to go all the way no matter what the cost"

But due to Windows not liking long filenames. it shorten to:
"no no im still not feeling it i want this achievement to have meant something it has to be a true reward for valiant effort i want to see some hustle stanley i want to see commitment a willingness to go"

**Comment by LUModder** [comment]
What about the hl2 dialogue of how could one man slipped through time and
all of that, that's pretty long.

**Comment by Tenrys** [comment]
You put some fitting name for the file then change the quote in the list generated.

**Comment by LUModder** [comment]
Yeah, but preprocessor is gonna overwrite it.

**Comment by Tenrys** [comment]
Huh, you change it after you use the preprocessor.
Or do you mean in next usage? If so, I don't know.

**Comment by LUModder** [comment]
Next useage

**Comment by zxz41** [comment]
Maybe just make a temporary hotfix? like a _hotfix sound folder/list till preprocessor learns to ignore files that's already listed?

**Comment by PotcFdk** [comment]
The preprocessor already supports sound aliases, so you could use those. If you don't want to just alias the sounds but rather actually define a new sound name, I could implement that, too.

Usage of the sound aliases:
1. Create a "map.txt" in the sound set directory (e.g. `autoadd/misc`)
2. Add one or more lines of the format `source; target`. Example: `ps1startup.ogg` exists, but you want a new alias called `psx startup` -> the line would be `ps1startup; psx startup`. Lines that begin with `#` are ignored.
3. When the preprocessor generates a new list for that soundset, the aliases will be respected in the generated sound list.

**Comment by Tenrys** [comment]
I think that's what they want.
How would this work for folders? (ex: every sound in autoadd/tenrys/tenrys aliased to tennies)

---- PotcFdk wrote ----

The preprocessor already supports sound aliases, so you could use those. If you don't want to just alias the sounds but rather actually define a new sound name, I could implement that, too.

Usage of the sound aliases:
1. Create a "map.txt" in the sound set directory (e.g. `autoadd/misc`)
2. Add one or more lines of the format `source; target`. Example: `ps1startup.ogg` exists, but you want a new alias called `psx startup` -> the line would be `ps1startup; psx startup`. Lines that begin with `#` are ignored.
3. When the preprocessor generates a new list for that soundset, the aliases will be respected in the generated sound list.

---

**Comment by PotcFdk** [comment]
Works exactly the same, `tenrys; tennies`

**Comment by Tenrys** [comment]
Okay

---- PotcFdk wrote ----

Works exactly the same, `tenrys; tennies`

---

**Comment by PotcFdk** [comment]
Just releaved version 0.6.3. You can now optionally append `replace` to replace a sound(set) name in the generated list.  
tl;dr example:
`tenrys; tennies` will generate the list with an additional copy: `tennies` will be an exact copy of `tenrys`.
`tenrys; tennies; replace` will do what the above does, but additionally, it will remove `tenrys` from the list, so you're only left with `tennies`.

I'll assume this solves your issue. If not, feel free to comment.

**Comment by zxz41** [comment]
One last thing. Shouldn't this function be mentioned in "HOW TO ADD SOUNDS.txt" for future reference?

And if yes. Should I/we copy paste what you said?

> Usage of the sound aliases:
> 1. Create a map.txt in the sound set directory (e.g. autoadd/misc/map.txt)
> 2. Add one or more lines of the format source; target. Example: ps1startup.ogg exists, but you want a new alias called psx startup -> the line would be ps1startup; psx startup. Lines that begin with # are ignored.
> 3. When the preprocessor generates a new list for that sound set, the aliases will be respected in the generated sound list.

**Comment by PotcFdk** [comment]
I guess that might make sense. You can use what I said, if you want.
